src/__tests__/integration/settings.test.js

```diff
@@ -0,0 +1,54 @@
+import { integrationTest, waitFor, uploadFile, getNextStepButton, getTestFile } from './index';
+import { act } from 'react-dom/test-utils';
+import React from 'react';
+import Settings from '../../components/layouts/pages/settings/Settings';
+
+const TEST_FILE = getTestFile();
+const PROFILE_VALUE = 'TAGGED_PDF';
+
+const isFileUploaded = state => state.pdfFiles.length;
+
+const getProfileValue = store => store.getState().jobSettings.profile;
+const simulateProfileChange = (component, value) =>
+    component
+        .find('#jobProfile')
+        .at(0)
+        .props()
+        .onChange({ target: { value } }, null);
+
+describe('Settings', () => {
+    it(
+        'Settigns page loaded',
```

Comment:
Page loaded should become describe of the following cases:
```
describe('Settings page loaded', () => {
    describe('No saved settings', () => {
        it('Profiles list not yet loaded')
        it('Profiles list loaded')
        it('Profiles list failed to load')
    });
    describe('Saved settings present', () => {
        // here we assume that profiles list loaded
        it('Saved settings restored')
        it('Saved settings contain invalid profile')
        it('Saved settings is not a valid json')
    });
})
```
If you can come up with other cases feel free to add them as well. The goal is to cover all code branches, you can consult with code coverage for that:
![image](https://user-images.githubusercontent.com/5430498/79750968-7dbabd80-831a-11ea-9c01-5be54a89982f.png)

Reply:
The latter section indeed is impossible to test in integration since `getDefaultState` is called only once for all the tests during first module import.

Tested that functionality with simpler unit tests in my fork.